src/core/auth/qgsauthmanager.cpp

```diff
@@ -103,6 +103,7 @@ QgsAuthManager *QgsAuthManager::instance()
 QgsAuthManager::QgsAuthManager()
 {
   mMutex = new QMutex( QMutex::Recursive );
+  mMasterPasswordMutex = new QMutex( QMutex::Recursive );
```

Comment:
I think this is leaked?
Use a unique_ptr instead?

Reply:
It leaks, like the pre-existing mMutex, but assumed it was fine since we're only running one instance until app is closed.

Reply:
We won't crash the system, but it's better to clean things up so tools like valgrind give cleaner reports.

Reply:
If you want, I can push an extra commit moving both the new and pre existing mutex to use unique_ptr.

Reply:
Yes please!